pkg/operator/ceph/cluster/rbd/mirror.go

```diff
@@ -96,19 +98,29 @@ func (m *Mirroring) Start() error {
 		}
 
 		// Start the deployment
-		deployment := m.makeDeployment(daemonConf)
-		if _, err := m.context.Clientset.Apps().Deployments(m.Namespace).Create(deployment); err != nil {
+		d := m.makeDeployment(daemonConf)
+		if _, err := m.context.Clientset.Apps().Deployments(m.Namespace).Create(d); err != nil {
 			if !errors.IsAlreadyExists(err) {
 				return fmt.Errorf("failed to create %s deployment. %+v", resourceName, err)
-				/* TODO: update rbd mirrors if already exist */
 			}
-			logger.Infof("%s deployment already exists", resourceName)
+			logger.Infof("deployment for rbd-mirror %s already exists. updating if needed", resourceName)
+			if _, err := updateDeploymentAndWait(m.context, d, m.Namespace); err != nil {
+				// fail could be an issue updating label selector (immutable), so try del and recreate
```

Comment:
Should `k8sutil.UpdateDeploymentAndWait()` handle this case of failure due to the label selector? And when would this failure happen? Since we're switching to the apps v1 it would happen once, or it will only happen if we change the label selector? In that case, why would we change the label selector?

Reply:
You know, I have no idea. I suppose this could be added into `k8sutil.UpdateDeploymentAndWait()`, but I'm not sure if it's needed for other daemons. If this is the case, I think I'd rather do that in a different PR.

I am changing the label selector here to use the consistent `rbd-mirror` wording instead of various instances of both `rbd-mirror` and `rbdmirror` on resources (and labels) created by Rook.

Reply:
Ok, so you saw this specifically because the label was being renamed in this case. That's pretty annoying that the update fails for that reason.